mods/default/nodes.lua

```diff
@@ -2013,10 +2013,12 @@ function default.register_chest(name, d)
 			local meta = minetest.get_meta(pos)
 			meta:set_string("formspec", nil)
 			local inv = meta:get_inventory()
-			local list = inv:get_list("main")
-			inv:set_list("main", nil)
-			inv:set_size("default:chest", 8*4)
-			inv:set_list("default:chest", list)
+			local list = inv:get_list("default:chest")
+			if list then
+				inv:set_size("main", 8*4)
+				inv:set_list("main", list)
+				inv:set_list("default:chest", nil)
+			end
```

Comment:
What I mean is that each LBM runs only once for each node. Thus, is code had upgraded the chests, then the list name will have been changed an the new LBM will not run to change the name back.